### PR TITLE
Raven db 13831 2

### PR DESCRIPTION
**Pull request by maximburyak**

### CODE REVIEW

File: src/Raven.Server/Documents/Subscriptions/SubscriptionStorage.cs [hunk 1]
```diff
@@ -460,5 +463,43 @@ public void ReleaseSubscriptionsSemaphore()
         {
             _concurrentConnectionsSemiSemaphore.Release();
         }
+
+        internal void CleanupSubscriptions()
+        {
+            var maxTaskLifeTime = (PlatformDetails.Is32Bits || this._db.Configuration.Storage.ForceUsing32BitsPager
+                        ? TimeSpan.FromHours(12)
```
Comment (ayende) [inline src/Raven.Server/Documents/Subscriptions/SubscriptionStorage.cs:469]: I don't think it matters, to be frank. Either way is very long

File: src/Raven.Server/Documents/Subscriptions/SubscriptionStorage.cs [hunk 2]
```diff
@@ -460,5 +463,43 @@ public void ReleaseSubscriptionsSemaphore()
...
+            foreach (var state in _subscriptionConnectionStates)
+            {
+                if (state.Value.Connection != null)
+                    continue;
+
+                var recentConnection = state.Value.MostRecentEndedConnection();
+                if (recentConnection == null)
+                    return;
+
+                if (recentConnection.Stats.LastMessageSentAt < oldestPossibleIdleSubscription)
```
Comment (ayende) [inline src/Raven.Server/Documents/Subscriptions/SubscriptionStorage.cs]: You are removing just one here, what if we have a lot of them? Should't we put this in a while?

File: src/Sparrow/Json/JsonOperationContext.cs [hunk 3]
```diff
@@ -325,6 +325,7 @@ public JsonOperationContext(int initialSize, int longLivedSize, SharedMultipleUs
                 TryDispose(_documentBuilder);
                 TryDispose(_arenaAllocator);
                 TryDispose(_arenaAllocatorForLongLivedValues);
+                TryExecute(_allocateStringValues.Clear);
```
Comment (ayende) [inline src/Sparrow/Json/JsonOperationContext.cs:328]: shouldn't we also set it to null?

File: src/Raven.Server/Documents/Subscriptions/SubscriptionStorage.cs (hunk 1 above)
Comment (maximburyak) [reply src/Raven.Server/Documents/Subscriptions/SubscriptionStorage.cs:469]: Yeah, I didn't want to add a new configuration here, so I decided to have a small preemptive move

**Review by ayende — COMMENTED** [review]
I assume you verified that in the long living contexts, we aren't hanging on to instances after the reset?

**Review by ayende — APPROVED** [review]

**Comment by maximburyak** [comment]
@ayende that's why I didn't touch bulk insert and smuggler, it's much trickier there..

**Comment by arekpalinski** [comment]
@maximburyak Could you check failing tests

**Comment by ppekrol** [comment]
@maximburyak can you take a look at whitespace issue also?

**Comment by maximburyak** [comment]
pushing a rebase

File: src/Raven.Server/Documents/Subscriptions/SubscriptionStorage.cs (hunk 2 above)
Comment (maximburyak) [reply src/Raven.Server/Documents/Subscriptions/SubscriptionStorage.cs]: done

File: src/Sparrow/Json/JsonOperationContext.cs (hunk 3 above)
Comment (maximburyak) [reply src/Sparrow/Json/JsonOperationContext.cs:328]: IMO, clear is enough here, we don't want to expose ourselves to potential NREs

**Pull request merged**